apis/r/R/TileDBArray.R

```diff
@@ -24,12 +24,15 @@ TileDBArray <- R6::R6Class(
       private$.mode <- mode
       if (is.null(private$tiledb_timestamp)) {
         spdl::debug("[TileDBArray::open] Opening {} '{}' in {} mode", self$class(), self$uri, mode)
-        tiledb::tiledb_array_open(self$object, type = mode)
+        private$.tiledb_array <- tiledb::tiledb_array_open(self$object, type = mode)
```

Comment:
I am a bit puzzled that we got by so far without an explicit copy and that this requires it now.

Reply:
The problem I'd hit without this reassignment seemed to be specific to setting WRITE timestamp, which our tests (in this PR) are not doing. But I added this based on the discussion on the TileDB-R thread and to be consistent with TileDBGroup.R (still not sure if this change to TileDBArray.R had gotten lost in rebase or smth)